### ISSUE TITLE
Caching one specific project folder with mutagen results in "Error" state.

### ISSUE THREAD

**Issue by Strayer** [docker/for-mac#4613]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 832DC137-D04E-4DE8-9132-345720B937E8/20200527180521

### Expected behavior

Caching of folder should succeed and state should be "Ready".

### Actual behavior

It starts synchronizing the folder and then simply states "Error".

### Information

  - macOS Version: 10.15.4 (19E287)

![grafik](https://user-images.githubusercontent.com/310624/83056517-db4ad400-a055-11ea-8852-06cdcbd1a9a8.png)

I'd love to provide more info, but apart from submitting a diagnostics report I don't know where to look.

### Steps to reproduce the behavior

Not sure I can describe how to reproduce the problem except by providing the project source code, which is obviously not possible. It is a simple Rails project with a webpack subproject, nothing special.

**Comment by Vmadmax** [comment]
Same issue here on a laravel project.
Diagnostic ID: C4F39822-496A-4F51-B5C7-A2531F03FAD7/20200527182630
macOS: 10.15.5 (19F96)

**Comment by Strayer** [comment]
I should add that a different project folder works fine, so it seems to be something in this project folder causing the error. Without any log files it is hard to say what exactly is going wrong.

**Comment by joostverkaik** [comment]
@Strayer did the caching process "restart" several times before it entered the Error state? I've seen it reach (near) 100% for several times now and then start again from 0%. I've read that UX is bad at the moment and there are no logs so there's no way of knowing what happens.

**Comment by Strayer** [comment]
No, it says "Scanning files ..." and then "Error", nothing in between. The other project folder says "Scanning files ..." and then "Ready".

**Comment by vitobotta** [comment]
Same problem here :( I have tried restarting a few times but it always goes into Error state.

**Comment by vitobotta** [comment]
Found the problem, it's symlinks. Removed a couple of symlinks and all is good now.

**Comment by mbentley** [comment]
I was trying it out myself and ran into the same issue and found https://github.com/docker/for-mac/issues/4595#issuecomment-634237214 where @djs55 pointed to a build that modifies the behavior and it worked as expected for me.

As a side note, I am seeing impressive performance; did a test build of a live docs preview app and a build that took ~55 seconds before was less than 30 seconds which gives me about the same performance as a bare metal linux machine running the same process in Docker with bind mounts.

**Comment by vitobotta** [comment]
I am also seeing big improvements, my tests run a lot faster now!

**Comment by Strayer** [comment]
> Found the problem, it's symlinks. Removed a couple of symlinks and all is good now.

That may be my issue too, since the project has some symlinks in .git for shared git hooks.

**Comment by joostverkaik** [comment]
I have also removed really big files and symlinks and it seems to be working now.

**Comment by guille-moe** [comment]
I tried the same edge version and the build from https://github.com/docker/for-mac/issues/4595#issuecomment-634237214
The shared folder display `Caching in progress (x.x%)` and when it end, it display `Error` then it loop back to `Scanning files ...` (then Caching, Error, etc...)

Not sure if this is relevant but to gather informations I did the command: 
`curl -X GET --unix-socket ~/Library/Containers/com.docker.docker/Data/docker-api.sock http:/localhost/cache/state`

Which returned:
`{"/Users/redacted/Projects/Stormz":{"status":"Error","ready":false,"problems":[": unable to ensure path does not exist: path exists"]}}`

**Comment by rpsu** [comment]
#4667 

Run diagnostics, extract the created zip file and look for a file `sync.json` - it lead me to the correct fix (symlink inside the sync folder)

**Comment by rpsu** [comment]
https://github.com/docker/for-mac/issues/4633#issuecomment-636906048 has even quicker way to see where the problem in caching lies
